api/index.js

```diff
@@ -7,52 +7,136 @@ var validate = require('jsonschema').validate;
     }
 
     console.log(`sending configuration: ${util.inspect(result)}`);
     res.json(result);
+  } catch (err) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+      error: err.message
+    });
   }
-  catch (err) {
-    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({ error: err.message });
+});
+
+app.post('/attachment', async(req, res) => {
+  if (!req.headers['user-token']) {
+    return res.status(HttpStatus.BAD_REQUEST).json({
+      error: `user-token request header is missing`
+    });
+  }
+
+  var userName = getUserId(req.headers['user-token']);
+  try {
+    var attachmentProcessingDetails = [];
+    var exch = new ews.ExchangeService(ews.ExchangeVersion.Exchange2013);
+    exch.Url = new ews.Uri(req.body.ewsUrl);
+    exch.Credentials = new ews.OAuthCredentials(req.body.attachmentToken);
+
+    var attachmentIds = req.body.attachments.map(a => a.id);
+
+    // NOTE: exch.GetAttachments() returns response with all the attachments contents in BASE64 strings. Therefore, attachments in big sizes might cause long response times over network, or high memory usage in the client side.
+    var response = await exch.GetAttachments(attachmentIds, ews.BodyType.Text, null);
+    var azureBlobService = azureStorage.createBlobService(azureStorageConnectionString);
+
+    await utils.callAsyncFunc(azureBlobService, 'createContainerIfNotExists', CONTAINER_NAME);
+
+    // Handle responses (for every attachemnt there is a response into reponses:
+    for (var i = 0; i < response.responses.length; i++) {
+      var fileName = response.responses[i].attachment.name;
+      var base64Content = response.responses[i].attachment.base64Content;
+      var binaryData = Buffer.from(base64Content, 'base64');
+      var contentHash = sha256(binaryData);
+      var blobName = userName + "/" + encodeURIComponent(contentHash) + "/" + fileName;
+
+      var binaryStream = intoStream(binaryData);
+
+      if (req.body.upload) {
+        await utils.callAsyncFunc(azureBlobService, 'createBlockBlobFromStream', CONTAINER_NAME, blobName, binaryStream, binaryData.length);
+        var sasToken = getSAS("attachments", azureBlobService, {
+          name: blobName
+        });
+        var sasUrl = azureBlobService.getUrl("attachments", blobName, sasToken, true);
+      }
+
+      attachmentProcessingDetails.push({
+        name: fileName,
+        hash: contentHash,
+        sasUrl: sasUrl
+      });
+    }
+
+    res.json({
+      attachmentsProcessed: attachmentProcessingDetails.length,
+      attachmentProcessingDetails: attachmentProcessingDetails
+    });
+  } catch (err) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+      error: err.message
+    });
   }
+
 });
 
-app.put('/proof', async (req, res) => {
-  
+function getSAS(CONTAINER_NAME, blobSvc, opts) {
+  var sharedAccessPolicy = {
+    AccessPolicy: {
+      Start: azureStorage.date.daysFromNow(-1),
+      Expiry: azureStorage.date.daysFromNow(7),
```

Comment:
The file is being downloaded immediately so no actual reason for more than just a few minutes...
try to see if there's date.minutesFromNow so it will only be valid for 1 minute.

Reply:
Changed to 2 minutes